beetmoverscript/script.py

```diff
@@ -31,12 +32,15 @@
     context.task = get_task(context.config)  # e.g. $cfg['work_dir']/task.json
     # 2. validate the task
     validate_task_schema(context)
-    # 3 grab manifest props with all the useful data
+    # 3 prepare manifest props file
+    #   a. grab manifest props with all the useful data
+    #   b. amend its platform field to proper one
     context.properties = await get_props(context)
+    amend_props(context, PLATFORM_MAP)
```

Comment:
nit: I'm trying to get myself in the habit of writing pure code with no side-effects. could you make amend_props a non mutating method?

e.g. `context.properties = update_props(context.properties)` where in update_props body impl you deepcopy context.properties and return the updated copy